### ISSUE TITLE
Can't remove reCaptcha screen in android mobile devices during Phone Authentication.

### ISSUE THREAD

**Issue by pkitatta** [firebase/firebase-android-sdk#2477]
Step 1: Are you in the right place?
Yes

 Step 2: Describe your environment

  * Android Studio version: 3.2.1
  * Firebase Component: firebase_auth
  * Component version: : ^0.18.4+1
  * Android device:  Fero A4001 Plus, Android version 6.0
  * flutter doctor
      Doctor summary (to see all details, run flutter doctor -v):
      [√] Flutter (Channel stable, 1.22.5, on Microsoft Windows [Version 10.0.18362.1256], locale en-US)
      
      [√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
      [√] Android Studio (version 3.2)
      [!] Connected device
          ! No devices available
      
      ! Doctor found issues in 1 category.

Step 3: Describe the problem
Phone auth has been working well in mobile devices, even in emulator, without launching recapture screen. Recently when I upgraded flutter and all the dependencies, phone auth started launching recapture browser in all devices.

The whole recaptcha process is a bit of an inconvenience and beats the simplicity of why we choose phone auth in the first place.
Recaptcha also prevents the normal authentication on phone with RAM that is below 1GB. This is because every time the browser is launched, the flutter app is pushed and killed in the background, and this cycle keeps repeating without letting the user signin.

I have followed the instructions in this video: https://www.youtube.com/watch?v=opBwTxicw1k, but I have failed to remove remove recapture on mobile devices.
Steps in video:

Add SHA-256
Enable Android Device Verification API in Project
I also get this in the console: SafetyNet Attestation fails basic integrity.

 Relevant Code:

```
https://github.com/Amanullahgit/flutter_phone_authentication/blob/main/lib/otp.dart
```

**Comment by Vitor-Bukovitz** [comment]
> Recaptcha also prevents the normal authentication on phone with RAM that is below 1GB. This is because every time the browser is launched, the flutter app is pushed and killed in the background, and this cycle keeps repeating without letting the user signin.

This part is really serious, Firebase Phone Auth should be compatible with any kind of device.

**Comment by malcolmdeck** [comment]
I can address this in several parts. 

- New versions of the SDK starting with 20.0.0 have, indeed, changed the behavior of Phone Authentication. This was done for several reasons, but the largest one was to allow for support of devices that do not have Google Play services. This is an expansion, not a contraction, of availability/compatibility. 
- The SafetyNet functionality closely mirrors the security practices present in previous version of the SDK; if it's failing for basic_integrity reasons, it should have been failing in the past. If you have a detailed report of the state of the phone which passes for old SDK versions but not for the new SafetyNet-based flows (and in particular details around why it fails basic_integrity), please report back with that information. We may have missed an edge case in comparing our security measures, and I'll happily look into fixing that. 
- I understand that the reCAPTCHA flow is cumbersome and may be too heavyweight for some clients. We currently don't have a better solution that meets our security requirements. 

I hope that helps. 
~Malcolm

**Comment by pkitatta** [comment]
@malcolmdeck thanks for taking the time to response, however everything is back to normal now. Let me expain

**IMPORTANT UPDATE**

As I was troubleshooting something else unrelated yesterday, I realized that the recaptcha screen had stopped launching. What did I do?

1. I changed the `compileSdkVersion` and` targetSdkVersion` app/build.gradle from 29 to 30.
2. `distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip` to `distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip` In the graddle-wrapper.properties. This is because I was using an new flutter app and the former comes by default.
3. I upgraded all the pabspec.yaml dependencies to their latest versions on pub.dev - and that includes `firebase_auth` and other firebase packages.

So today when I was running the app on emulator I realised the I didn't see the recaptcha screen. When I run the app on 512MB memory, android 6 phones the results were the same - no recaptcha screen.

I am not the most advanced of developers to know what really happened but am hoping you guys can deduce where the possible pressure point was and advice others precisely.

As for now am the happiest man - No More recaptcha screens for my clients!
